actions/checkForAndPlaceOrder.ts

```diff
@@ -161,7 +161,18 @@ const _checkForAndPlaceOrder: ActionFn = async (
 
       // Don't try again the same order, in case thats the poll result
       if (pollResult.result === PollResultCode.DONT_TRY_AGAIN) {
-        ordersPendingDelete.push(conditionalOrder);
+        // Check if the TX exists. This avoids one race condition where the order has been registered but our RPC don't see it yet.
+        // It can happen if:
+        //    - Tenderly RPC node is ahead of our RPC node
+        //    - There's a reorg between the registration and the polling
+        const existTransaction = await chainContext.provider
+          .getTransaction(conditionalOrder.tx)
+          .then(() => true)
+          .catch(() => false);
```

Comment:
I'm wondering: Is it possible by any chance that it returns undefined when the tx doesn't exist rather than throwing an exception?

Reply:
I don't want it to throw an exception, i want to:

- existTransaction be true if I know for sure it exists
- be false other wise (if it doesn't exist, or even if Im unsure, like in an unhandled exception)

I use this to avoid deletion of the order. I want to make sure we know what we are doing

Reply:
That's my question.
Can `getTransaction` return something that's falsy without throwing?
The result from `getTransaction` is being ignored and it returns `true` regardless of the value.